ResearchKit/Common/ORKLocationSelectionView.m

```diff
@@ -0,0 +1,343 @@
+        [self removeConstraint:_textFieldBottomConstraint];
+        _textFieldBottomConstraint = [NSLayoutConstraint constraintWithItem:_textField attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:_mapView attribute:NSLayoutAttributeTop multiplier:1.0 constant:0.0];
+        [self addConstraint:_textFieldBottomConstraint];
+        _mapViewHeightConstraint = [NSLayoutConstraint constraintWithItem:_mapView attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:nil attribute:NSLayoutAttributeNotAnAttribute multiplier:1.0 constant:0.0];
+        [self addConstraint:_mapViewHeightConstraint];
+        [self addConstraint:[NSLayoutConstraint constraintWithItem:_mapView attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0.0]];
+        
+        [self layoutIfNeeded];
+        _mapViewHeightConstraint.constant = 228.0;
+        __weak __typeof__(self) weakSelf = self;
+        [UIView animateWithDuration:0.3 animations:^{
+            [weakSelf layoutIfNeeded];
+            
+            if ([weakSelf.delegate respondsToSelector:@selector(selectionViewNeedsResize:)]) {
+                [weakSelf.delegate selectionViewNeedsResize:self];
+            }
+        }];
+    }
+}
+
+- (void)hideMapViewAnimated:(BOOL)animated {
+    if (!_mapView) {
+        return;
+    }
+    
+    if (!animated) {
+        [_mapView removeFromSuperview];
+        _mapView = nil;
+        _mapViewHeightConstraint = nil;
+        _textFieldBottomConstraint = [NSLayoutConstraint constraintWithItem:_textField attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:self attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0.0];
+        [self addConstraint:_textFieldBottomConstraint];
+        [self layoutIfNeeded];
+    } else {
+        [self layoutIfNeeded];
+        _mapViewHeightConstraint.constant = 0.0;
+        __weak __typeof__(self) weakSelf = self;
+        [UIView animateWithDuration:0.3 animations:^{
+            [weakSelf layoutIfNeeded];
+        } completion:^(BOOL finished) {
+            [weakSelf.mapView removeFromSuperview];
+            weakSelf.mapView = nil;
+            weakSelf.mapViewHeightConstraint = nil;
+            weakSelf.textFieldBottomConstraint = [NSLayoutConstraint constraintWithItem:weakSelf.textField attribute:NSLayoutAttributeBottom relatedBy:NSLayoutRelationEqual toItem:weakSelf attribute:NSLayoutAttributeBottom multiplier:1.0 constant:0.0];
+            [weakSelf addConstraint:weakSelf.textFieldBottomConstraint];
+            [weakSelf layoutIfNeeded];
+            if ([weakSelf.delegate respondsToSelector:@selector(selectionViewNeedsResize:)]) {
+                [weakSelf.delegate selectionViewNeedsResize:self];
+            }
+        }];
+    }
+}
```

Comment:
The animation to show and hide the map view is very unstable. It distracts the user's focus from the point they tapped on the screen to another point on the screen. It would be best to animate the cell, so that the user's focus can stay at the point that they tapped in order to enter their location.